File: src/main/java/com/learnershi/rclasssocket/repository/SessionUserRepository.java
package com.learnershi.rclasssocket.repository;

import org.springframework.stereotype.Repository;

import java.util.Map;
import java.util.concurrent.ConcurrentHashMap;

@Repository
public class SessionUserRepository {
    record UserSession(String sessionId, String userSeq, String classRoomId) {}

    private static Map<String, UserSession> userSessionMap;

    public SessionUserRepository() {
        if (userSessionMap == null) {
            userSessionMap = new ConcurrentHashMap<>();
        }
    }

    public void addSession(String sessionId, String userSeq, String classRoomId) {
        userSessionMap.put(sessionId, new UserSession(sessionId, userSeq, classRoomId));
    }

    public void removeSession(String sessionId) {
        userSessionMap.remove(sessionId);
    }

    public String getUserSeq(String sessionId) {
        return userSessionMap.get(sessionId).userSeq;
    }

    public String getClassRoomId(String sessionId) {
        return userSessionMap.get(sessionId).classRoomId;
    }

}

File: src/main/java/com/learnershi/rclasssocket/repository/SessionUserRepository.kt
package com.learnershi.rclasssocket.repository

import org.springframework.stereotype.Repository
import java.util.concurrent.ConcurrentHashMap

@Repository
class SessionUserRepository {
    @JvmRecord
    internal data class UserSession(val sessionId: String, val userSeq: String, val classRoomId: String)

    init {
        if (userSessionMap == null) {
            userSessionMap = ConcurrentHashMap()
        }
    }

    fun addSession(sessionId: String, userSeq: String, classRoomId: String) {
        userSessionMap!![sessionId] = UserSession(sessionId, userSeq, classRoomId)
    }

    fun removeSession(sessionId: String) {
        userSessionMap!!.remove(sessionId)
    }

    fun getUserSeq(sessionId: String): String {
        return userSessionMap!![sessionId]!!.userSeq
    }

    fun getClassRoomId(sessionId: String): String {
        return userSessionMap!![sessionId]!!.classRoomId
    }

    companion object {
        private var userSessionMap: MutableMap<String, UserSession>? = null
    }
}